File: spec/java/src/io/kaitai/struct/spec/TestDebugEnumName.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.DebugEnumName;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
public class TestDebugEnumName extends CommonSpec {

    @Test
    public void testDebugEnumName() throws Exception {
        DebugEnumName r = DebugEnumName.fromFile(SRC_DIR + "fixed_struct.bin");

        // --debug implies --no-auto-read
        r._read();

        assertEquals(r.one(), DebugEnumName.TestEnum1.ENUM_VALUE_80);
        assertEquals(r.arrayOfInts().get((int) 0), DebugEnumName.TestEnum2.ENUM_VALUE_65);
        assertEquals(r.testType().field1(), DebugEnumName.TestSubtype.InnerEnum1.ENUM_VALUE_67);
        assertEquals(r.testType().instanceField(), DebugEnumName.TestSubtype.InnerEnum2.ENUM_VALUE_11);
    }
}

File: spec/python/spec/test_debug_enum_name.py
import unittest
from testformats.debug_enum_name import DebugEnumName
from testformats import debug_enum_name

class TestDebugEnumName(unittest.TestCase):
    def test_debug_enum_name(self):
        with DebugEnumName.from_file('src/fixed_struct.bin') as r:
            # --debug implies --no-auto-read
            r._read()

            self.assertEqual(r.one, debug_enum_name.DebugEnumName.TestEnum1.enum_value_80)
            self.assertEqual(r.array_of_ints[0], debug_enum_name.DebugEnumName.TestEnum2.enum_value_65)
            self.assertEqual(r.test_type.field1, debug_enum_name.DebugEnumName.TestSubtype.InnerEnum1.enum_value_67)
            self.assertEqual(r.test_type.instance_field, debug_enum_name.DebugEnumName.TestSubtype.InnerEnum2.enum_value_11)
